middleware/authorization.middleware.ts

```diff
@@ -6,10 +6,19 @@ import {
   verifyToken
 } from "../shared/jwt-token.helpers";
 
+const ignoreAuthRoutes = [
+  "auth",
+  "farms"
+];
+
 // all requests pass through this function to check "Authorization" token
 async function authUser (req: Request, res: Response, next: NextFunction) {
-  // auth paths, security not needed
-  if (req.url.includes("auth")) {
+  // check for ignore paths
+  const isIgnorePath = ignoreAuthRoutes.some(route => req.url.split("/")?.at(3)?.toLocaleLowerCase("en-US") === route);
+  
```

Comment:
If params had "?auth" the validation was true, This was a major flaw which could bypass security checks

Reply:
This has been resolved by taking the exact initial URL route to match and ignore certain routes